mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java

```diff
@@ -3432,13 +3434,16 @@ private void showBrowserSearch() {
         // We do this here because there are glitches when unlocking a device with
         // BrowserSearch in the foreground if we use BrowserSearch.onStart/Stop.
         getWindow().setBackgroundDrawableResource(android.R.color.white);
+        /* Cliqz end */
     }
 
     private void hideBrowserSearch() {
         /* Cliqz start */
-        // if (!mBrowserSearch.getUserVisibleHint()) {
-        //     return;
-        // }
+        if (!mPreferenceManager.isQuickSearchEnabled()) {
+            if (!mBrowserSearch.getUserVisibleHint()) {
+                return;
```

Comment:
You should merge both the "if" into one statement

Reply:
done